### ISSUE TITLE
[FEA] Unable to install cuDF on TX2

### ISSUE THREAD

**Issue by thedatatechie** [rapidsai/cudf#2505]
**Describe the bug**
cuDF setup is failing on Nvidia TX2 dev kit. Docker image also not working.

**Steps/Code to reproduce bug**
1. Take a Nvidia TX2 Dev kit, Pascal architecture
2. Flash it using Jetpack 4.2.1, Ubuntu 18.04, CUDA 10.0
3. Pull Docker image 
    sudo docker pull rapidsai/rapidsai:0.8-cuda10.0-runtime-ubuntu18.04-gcc7-py3.6
4. Start Docker container
    sudo docker run --runtime=nvidia --rm -it -p 8888:8888 -p 8787:8787 -p 8786:8786 rapidsai/rapidsai:0.8-cuda10.0-runtime-ubuntu18.04-gcc7-py3.6

Docker container fails to start complaining cuda-compat and cuda-cudart libraries doesn't exist. I could find them for AMD but not for ARM.
Equivalent for AMD,
cuda-compat-10-0_410.104-1_amd64.deb
cuda-cudart-10-0_10.0.130-1_amd64.deb

The situation is same, if I try to install it from the master branch in git repo.

**Expected behavior**
throws an error/exception about the coda-compat library.

**Environment overview (please complete the following information)**
Nvidia TX2, Pascal Architecture, Ubuntu18.04, CUDA 10.0
- Docker image "rapidsai/rapidsai:0.8-cuda10.0-runtime-ubuntu18.04-gcc7-py3.6"

- sudo docker pull rapidsai/rapidsai:0.8-cuda10.0-runtime-ubuntu18.04-gcc7-py3.6
- sudo docker run --runtime=nvidia --rm -it -p 8888:8888 -p 8787:8787 -p 8786:8786 rapidsai/rapidsai:0.8-cuda10.0-runtime-ubuntu18.04-gcc7-py3.6

**Comment by kkraus14** [comment]
The Docker image is built with x86 binaries currently so that is expected. Compiling cudf for TX2 / ARM is being explored in #2593.

**Comment by kkraus14** [comment]
cc @awthomp who's been exploring this

**Comment by awthomp** [comment]
Building cuDF on the TX2 is still a WIP, but here's an update and what I've done so far:

## Conda For ARM
* Install [conda4aarch64](https://github.com/jjhelmus/conda4aarch64/releases).
* Add c4aarch64 and conda-forge channels to conda configuration
```
$ conda config --add channels c4aarch64
$ conda config --add channels conda-forge
```

# cuDF
## Conda Dependencies
```
$ conda install -c numba numba
$ conda install cmake_setuptools boost-cpp cython pandas cffi rapidjson \
  double-conversion flatbuffers
```

## Pip Dependencies
```
$ pip install cmake-setuptools
```

## GitHub Dependencies
### DLPack
```
$ git clone https://github.com/dmlc/dlpack.git
$ cd dlpack
$ mkdir build && cd build
$ cmake .. -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX
$ make -j6
$ make install
```

## Install Directions
### RMM
```
$ git clone --recurse-submodules https://github.com/rapidsai/rmm.git
$ cd rmm
$ mkdir build && cd build
$ cmake .. -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX
$ make -j6
$ make install
```

### cuStrings
```
$ git clone --recurse-submodules https://github.com/rapidsai/custrings.git
$ cd custrings
$ ./build.sh
```

### libcudf
Note: I had to edit [ConfigureArrow](https://github.com/rapidsai/cudf/blob/master/cpp/cmake/Modules/ConfigureArrow.cmake) to enable building with Python since pyarrow is not available in the available conda channels. Set: `DARROW_PYTHON=ON`, `DARROW_COMPUTE=ON`, and `DARROW_BUILD_SHARED=ON`

Note: I could not get a successful build of libcudf with parallel compilation, i.e. `make -j6` and edited the build.sh script to build with one core.

```
$ git clone --recurse-submodules https://github.com/rapidsai/cudf.git
$ cd cudf
$ ./build.sh libcudf
```
Install Python Arrow bindings
```
$ cd cpp/build/arrow/arrow/python
$ python setup.py build_ext --inplace
$ python setup.py install --single-version-externally-managed --record=record.txt
```
The next step is to install the Python bindings for cuDF via `./build.sh cudf`, but that currently gives an arrow error that it can't find the platform.h header file. Trying to track down the include files and what not presently.

**Comment by heavyinfo** [comment]
Based upon the work done by @awthomp, others and discussions with @kkraus14 reg the issue https://github.com/rapidsai/cudf/issues/4121; I was able to successfully build and install cudf on Jetson Nano and should work fine on TX2.

I have written the build procedure for cudf on ARM (Jetson) here - https://gist.github.com/heavyinfo/da3de9b188d41570f4e988ceb5d81abc.

**Comment by kkraus14** [comment]
Closed based on #4121. We are working on building conda packages in the future.
